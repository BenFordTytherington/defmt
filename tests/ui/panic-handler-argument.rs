#![no_main]
#![no_std]

use defmt_macros::info;
#[panic_handler]
fn foo(x: bool) -> ! { info!("{:?}", x); loop {} }
