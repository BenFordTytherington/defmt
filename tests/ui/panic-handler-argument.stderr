error[E0152]: found duplicate lang item `panic_impl`
 --> $DIR/panic-handler-argument.rs:6:1
  |
6 | fn foo(x: bool) -> ! { info!("{:?}", x); loop {} }
  | ^^^^^^^^^^^^^^^^^^^^
  |
  = note: the lang item is first defined in crate `std` (which `defmt` depends on)
  = note: first definition in `std` loaded from /Users/brigitte/.rustup/toolchains/stable-x86_64-apple-darwin/lib/rustlib/x86_64-apple-darwin/lib/libstd-518979da66993550.rlib
  = note: second definition in the local crate (`$CRATE`)

error: argument should be `&PanicInfo`
 --> $DIR/panic-handler-argument.rs:6:11
  |
6 | fn foo(x: bool) -> ! { info!("{:?}", x); loop {} }
  |           ^^^^
